TPCClusterFinder: Some code cleanup

fweig opened the pull request:
This PR includes some cleanup for the TPCClusterFinder:
* Replace `size_t` with 4 byte integers where possible.
* Simplify kernel names.
* Cleanup macros.
* Remove old, obsolete implementation.

davidrohr commented on GPU/Common/GPUDefGPUParameters.h:
```diff
@@ -285,15 +285,15 @@
 #define GPUCA_LB_GPUTPCCFDeconvolution GPUCA_LB_CLUSTER_FINDER
 #define GPUCA_LB_GPUTPCCFClusterizer GPUCA_LB_CLUSTER_FINDER
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanUpStart GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanUp GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanTop GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_nativeScanDown GPUCA_THREAD_COUNT_SCAN
-#define GPUCA_LB_GPUTPCCFStreamCompaction_compact GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step0 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step1 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step2 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step3 GPUCA_THREAD_COUNT_SCAN
+#define GPUCA_LB_GPUTPCCFStreamCompaction_step4 GPUCA_THREAD_COUNT_SCAN
```
Actually, why do you change this from the names to step[n] and then to integers below?
In my opinion, it was easier to read before.

fweig replied:
I thought it looked a bit cleaner. But it does hurt readability. I'll revert it.

davidrohr commented on GPU/GPUTracking/TPCClusterFinder/CfUtils.h:
```diff
@@ -77,8 +81,9 @@ class CfUtils
     tpccf::Delta2 d = neighbors[x + offset];
-    LOOP_UNROLL_ATTR for (unsigned int i = y; i < wgSize; i += (elems / N))
-    {
+
+    GPUCA_UNROLL(, U())
+    for (unsigned int i = y; i < wgSize; i += (elems / N)) {
```
If you want to unroll on AMD and NVIDIA, you should do GPUCA_UNROLL(U(), U())

fweig commented:
Looks like the CI didn't like the unrolling of that loop anyway. I think there are more places in the cluster finder where loop unrolling might be beneficial. But that's for another PR.

Fixed the naming of the kernels with the last commit and fixed a minor bug in the noise suppression where the wrong neighbors were checked for minima. I'll work on the gather bug now.

davidrohr commented:
@fweig : Unfortunately, this seems to break something. Running with my dataset o2-100 on asustest04, the clusterizer crashes after this PR, and it works again after I revert it. Can you have a look?
